packages/template-retail-react-app/app/hooks/use-selected-store.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useContext} from 'react'
+import {useStores} from '@salesforce/commerce-sdk-react'
+import {StoreLocatorContext} from '@salesforce/retail-react-app/app/contexts/store-locator-provider'
+
+/**
+ * Custom hook to get the selected store from StoreLocatorContext and retrieve store details
+ * @returns {Object} Object containing store data, loading state, and error state
+ */
+export const useSelectedStore = () => {
+    const context = useContext(StoreLocatorContext)
```

Comment:
Do we need to have a Provider for store locator? I am trying to shy away from using React Provider as much as possible. With the use of React query where it handles state and cache machenism, the need of provider is obsolate. React query will handle any duplicated calls to the api even if we use hook in multiple places. Please follow current pattern of use basket [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/template-retail-react-app/app/hooks/use-current-basket.js) if you want to have a hook that returns the current store locator.